tests/functional/object/mcg/test_s3_vector.py

```
@@ -0,0 +1,205 @@
+        )
+        index_name = f"test-vector-index-{random_suffix}"
+        dimension = 3
+        num_vectors = 100
+
+        # Step 1: Create a vector bucket using factory
+        # Factory auto-creates: nsfs namespacestore + vectorPolicy bucketclass
+        logger.info(
+            "Creating vector bucket (auto-creates nsfs + vectorPolicy bucketclass)"
+        )
+        vector_bucket_name = vector_bucket_factory(amount=1)[0].name
+        logger.info(f"✓ Vector bucket created: {vector_bucket_name}")
```

Ruff (0.15.13)

[warning] Logging statement uses f-string

(G004)